### ISSUE TITLE
Is it possible to get a 32-bit vst2 build for Windows?

### ISSUE THREAD

**Issue by Dinir** [werman/noise-suppression-for-voice#137]
There's a command to compile for x32 on the readme, but "cross-compiling for windows" is separately listed and only for x64.
I tried to run the cross-compiling command with having `mingw64` replaced by `mingw32`, which probably obviously doesn't work.

Is a 32-bit Windows build not supported?

**Comment by werman** [comment]
There is nothing preventing it from being built for win32. Probably you should build it with:

```
cmake -H. -Bbuild-rel-x32 -DCMAKE_BUILD_TYPE=Release -A Win32
cmake --build build-rel-x32 --config Release --parallel
```

I'm not 100% sure if it the right command to build x32 version with MSVC...

**Comment by Dinir** [comment]
It seems not. The commands failed in the first line.
```
-- Building for: NMake Makefiles
CMake Error at CMakeLists.txt:1 (project):
  Generator

    NMake Makefiles

  does not support platform specification, but platform

    Win32

  was specified.


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```

The output log simply says my system is `Windows - 10.0.19044 - AMD64`.

I installed CMake yesterday only to try to make a 32-bit build of this plugin, so it's 3.24.0 installed using the x64 installer provided by the homepage.

**Comment by werman** [comment]
What if you explicitly specify Visual Studio generator:

`cmake -H. -Bbuild-rel-x32 -G "Visual Studio 16 2019" -DCMAKE_BUILD_TYPE=Release -A Win32`

Note, you have to have MSVC installed.

**Comment by Dinir** [comment]
LUCKILY I happened to have Visual Studio installed.
I need to first find where I can find the string to put after `-G`.

**Comment by werman** [comment]
Then you could probably the project directly in Visual Studio and build it that way.

**Comment by Dinir** [comment]
I don't think I can do that. I don't have enough knowledge to make a whole new project out of existing codes that can actually succeed making a build out of them.

**Comment by werman** [comment]
Visual Studio has built-in support for cmake projects, so no creation of new project is necessary. 

Anyway, I think I have a Virtual Machine with Visual Studio, so I could check win32 compilation in a few days.

**Comment by werman** [comment]
Ok, tested in Virtual Machine and this was enough to build the lib:

```
cmake -Bbuild-rel-x32 -G "Visual Studio 16 2019" -DCMAKE_BUILD_TYPE=Release -A Win32
cmake --build build-rel-x32 --config Release --parallel
```

If cmake cannot find Visual Studio you could try (I'm using powershell):
```
cmd.exe /c "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvars32.bat"
```
before running cmake

**Comment by Dinir** [comment]
I had to install a new Visual Studio for this because Preview is different to Community. The build succeeded. I got one CMake Warning from the first line that says the manually-specified variable `CMAKE_BUILD_TYPE` was not used. But the build is seemingly working.

Here's the list of items I checked when installing VS Community to make it as small as possible, as I have no use for it other than building this plugin, and VS is huge.

Workload *Desktop & Mobile* -> *Desktop development with C++*
- MSVC v143
- Windows 10 SDK (10.0.19041.0)
- C++ CMake tools for Windows
